### PR TITLE
Add TiCDC Debezium Protocol docs

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? (Required)

Add a new doc "TiDB Debezium Protocol"

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/16513
- Other reference link(s): https://github.com/pingcap/tiflow/pull/10197, https://github.com/pingcap/tiflow/pull/10307

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-debezium.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
+---
+title: TiCDC Debezium Protocol
+summary: Learn the concept of the TiCDC Debezium Protocol and how to use it.
+---
+
+# TiCDC Debezium Protocol
+
+[Debezium](https://debezium.io/) is a tool for capturing database changes. It converts each captured database change into a message called an "event" and sends these events to Kafka. Starting from v8.0.0, TiCDC supports sending TiDB changes to Kafka in Debezium format, simplifying the data link of using Debezium.
```
Comment (benmeadowcroft) [inline ticdc/ticdc-debezium.md:8]: I am not sure what the phrase "simplifying the data link" is intended to convey here. I would suggest changing:

> Starting from v8.0.0, TiCDC supports sending TiDB changes to Kafka in Debezium format, simplifying the data link of using Debezium.

To something like:

> Starting from v8.0.0, TiCDC supports sending TiDB changes to Kafka using a Debezium style output format, simplifying migration from MySQL databases for users who had previously been using Debezium's MySQL integration.

File: ticdc/ticdc-debezium.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+| payload.before    | JSON   | The data value before the event statement change. For `"c"` events, the value of the `before` field is `null`.     |
+| payload.after     | JSON   | The data value after the event statement change. For `"d"` events, the value of the `after` field is `null`.     |
+| payload.source.commit_ts     | Number  | The `CommitTs` identifier when TiCDC generates this message.       |
+| payload.source.db     | String   | The name of the database where the event occurs.    |
+| payload.source.table     | String  |  The name of the data table where the event occurs.   |
+| schema.fields     | JSON   |  The type information of each field in the payload, including the schema information of the row data before and after the change.   |
+
+### Data type mapping
+
+The data format mapping in the TiCDC Debezium message basically follows the [Debezium data type mapping rules](https://debezium.io/documentation/reference/2.4/connectors/mysql.html#mysql-data-types), which is generally consistent with the native message of the Debezium Connector. However, for some data types, the following differences exist between TiCDC Debezium and Debezium Connector messages:
```
Comment (benmeadowcroft) [inline ticdc/ticdc-debezium.md:132]: I would suggest making it clear that we are following the Debezium type mappings used for MySQL compatibility. E.g. changing the first sentence to something like:

> The data format mapping in the TiCDC Debezium message basically follows the [Debezium data type mapping rules](https://debezium.io/documentation/reference/2.4/connectors/mysql.html#mysql-data-types), which is generally consistent with the native message of the Debezium Connector for MySQL.

**Review by benmeadowcroft — COMMENTED** [review]
I have added a couple of comments with some suggestions.

File: ticdc/ticdc-debezium.md (hunk 1 above)
Comment (lilin90) [reply ticdc/ticdc-debezium.md:8]: That's more clear.

**Comment by lilin90** [comment]
Thanks @benmeadowcroft! All your comments have been resolved.

File: ticdc/ticdc-debezium.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+summary: Learn the concept of the TiCDC Debezium Protocol and how to use it.
+---
+
+# TiCDC Debezium Protocol
+
+[Debezium](https://debezium.io/) is a tool for capturing database changes. It converts each captured database change into a message called an "event" and sends these events to Kafka. Starting from v8.0.0, TiCDC supports sending TiDB changes to Kafka using a Debezium style output format, simplifying migration from MySQL databases for users who had previously been using Debezium's MySQL integration.
+
+## Use the Debezium message format
+
+When you use Kafka as the downstream sink, specify the `protocol` field of `sink-uri` as `debezium`. Then TiCDC encapsulates the Debezium message based on the event and send TiDB data change events to the downstream.
```
Comment (lilin90) [inline ticdc/ticdc-debezium.md:12]:
```suggestion
When you use Kafka as the downstream sink, specify the `protocol` field as `debezium` in `sink-uri` configuration. Then TiCDC encapsulates the Debezium message based on the event and send TiDB data change events to the downstream.
```

File: ticdc/ticdc-debezium.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+}
+```
+
+The key fields of the preceding JSON data are explained as follows:
+
+| Field      | Type   | Description                                            |
+|:----------|:-------|:-------------------------------------------------------|
+| payload.op        | String | The type of the change event. `"c"` indicates an `INSERT` event, `"u"` indicates an `UPDATE` event, and `"d"` indicates a `DELETE` event.  |
+| payload.ts_ms     | Number | The timestamp (in milliseconds) when TiCDC generates this message. |
+| payload.before    | JSON   | The data value before the event statement change. For `"c"` events, the value of the `before` field is `null`.     |
```
Comment (qiancai) [inline ticdc/ticdc-debezium.md:123]:
```suggestion
| payload.before    | JSON   | The data value before the change event of a statement. For `"c"` events, the value of the `before` field is `null`.     |
```

File: ticdc/ticdc-debezium.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+| payload.after     | JSON   | The data value after the event statement change. For `"d"` events, the value of the `after` field is `null`.     |
```
Comment (qiancai) [inline ticdc/ticdc-debezium.md:124]:
```suggestion
| payload.after     | JSON   | The data value after the change event of a statement. For `"d"` events, the value of the `after` field is `null`.     |
```

File: ticdc/ticdc-debezium.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+# TiCDC Debezium Protocol
+
+[Debezium](https://debezium.io/) is a tool for capturing database changes. It converts each captured database change into a message called an "event" and sends these events to Kafka. Starting from v8.0.0, TiCDC supports sending TiDB changes to Kafka using a Debezium style output format, simplifying migration from MySQL databases for users who had previously been using Debezium's MySQL integration.
+
+## Use the Debezium message format
+
+When you use Kafka as the downstream sink, specify the `protocol` field as `debezium` in `sink-uri` configuration. Then TiCDC encapsulates the Debezium message based on the event and send TiDB data change events to the downstream.
+
+Currently, the Debezium protocol only supports Row changed events and directly ignores DDL events and WATERMARK events. A Row changed event represents a change in a row of data. When a row changes, the Row changed event is sent, including relevant information about the row both before and after the change.
```
Comment (qiancai) [inline ticdc/ticdc-debezium.md:14]:
```suggestion
Currently, the Debezium protocol only supports Row changed events and directly ignores DDL events and WATERMARK events. A Row changed event represents a data change in a row. When a row changes, the Row changed event is sent, including relevant information about the row both before and after the change. A WATERMARK event marks the replication progress of a table, indicating that all events less than the watermark have been sent to the downstream.
```

Comment (qiancai) [reply ticdc/ticdc-debezium.md:14]: Do we need to use "Row Changed Event" as [TiCDC Open Protocol | PingCAP Docs](https://docs.pingcap.com/tidb/dev/ticdc-open-protocol) does?

image

File: ticdc/ticdc-debezium.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+
+| Field      | Type   | Description                                            |
+|:----------|:-------|:-------------------------------------------------------|
+| payload.op        | String | The type of the change event. `"c"` indicates an `INSERT` event, `"u"` indicates an `UPDATE` event, and `"d"` indicates a `DELETE` event.  |
+| payload.ts_ms     | Number | The timestamp (in milliseconds) when TiCDC generates this message. |
+| payload.before    | JSON   | The data value before the event statement change. For `"c"` events, the value of the `before` field is `null`.     |
+| payload.after     | JSON   | The data value after the event statement change. For `"d"` events, the value of the `after` field is `null`.     |
+| payload.source.commit_ts     | Number  | The `CommitTs` identifier when TiCDC generates this message.       |
+| payload.source.db     | String   | The name of the database where the event occurs.    |
+| payload.source.table     | String  |  The name of the data table where the event occurs.   |
```
Comment (qiancai) [inline ticdc/ticdc-debezium.md:127]:
```suggestion
| payload.source.table     | String  |  The name of the table where the event occurs.   |
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: ticdc/ticdc-debezium.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+summary: Learn the concept of the TiCDC Debezium Protocol and how to use it.
+---
+
+# TiCDC Debezium Protocol
+
+[Debezium](https://debezium.io/) is a tool for capturing database changes. It converts each captured database change into a message called an "event" and sends these events to Kafka. Starting from v8.0.0, TiCDC supports sending TiDB changes to Kafka using a Debezium style output format, simplifying migration from MySQL databases for users who had previously been using Debezium's MySQL integration.
+
+## Use the Debezium message format
+
+When you use Kafka as the downstream sink, specify the `protocol` field as `debezium` in `sink-uri` configuration. Then TiCDC encapsulates the Debezium message based on the event and send TiDB data change events to the downstream.
```
Comment (qiancai) [inline ticdc/ticdc-debezium.md:12]:
```suggestion
When you use Kafka as the downstream sink, specify the `protocol` field as `debezium` in `sink-uri` configuration. Then TiCDC encapsulates the Debezium message based on the event and sends TiDB data change events to the downstream.
```

File: ticdc/ticdc-debezium.md (hunk 6 above)
Comment (lilin90) [reply ticdc/ticdc-debezium.md:14]: Although I don't think it's necessary to capitalize all words, we can keep its usage consistent among different docs pages.

**Review by hongyunyan — APPROVED** [review]

**Pull request merged**